appinventor/components/src/com/google/appinventor/components/runtime/util/NativeOpenStreetMapController.java

```diff
@@ -964,37 +964,43 @@ private void getMarkerDrawableVector(MapMarker aiMarker,
       }
     }
     final String markerAsset = aiMarker.ImageAsset();
-    if (markerAsset != null && markerAsset.length() != 0) {
+    Log.d(TAG, markerAsset + " Marker Asset");
+
+    if (markerAsset == null || markerAsset.length() == 0) {
       try {
-        markerSvg = SVG.getFromAsset(view.getContext().getAssets(), markerAsset);
-      } catch (SVGParseException e) {
-        Log.e(TAG, "Invalid SVG in Marker asset", e);
-      } catch (IOException e) {
-        Log.e(TAG, "Unable to read Marker asset", e);
-      }
-      if (markerSvg == null) {
-        // Attempt to retrieve asset from ReplForm storage location
-        InputStream is = null;
-        try {
-          is = MediaUtil.openMedia(form, markerAsset);
-          markerSvg = SVG.getFromInputStream(is);
-        } catch (SVGParseException e) {
-          Log.e(TAG, "Invalid SVG in Marker asset", e);
-        } catch (IOException e) {
-          Log.e(TAG, "Unable to read Marker asset", e);
-        } finally {
-          IOUtils.closeQuietly(TAG, is);
-        }
-      }
-    }
-    if (markerSvg == null) {
-      markerSvg = defaultMarkerSVG;
-    }
-    try {
-      callback.onSuccess(rasterizeSVG(aiMarker, markerSvg));
-    } catch(Exception e) {
-      callback.onFailure(e.getMessage());
+        // Passes the defaultMarkerSVG
+        Log.d(TAG, "PASSING DEFAULT before async");
+        callback.onSuccess(rasterizeSVG(aiMarker, defaultMarkerSVG));
+      } catch (Exception e) {
+        callback.onFailure(e.getMessage());
+      } 
+      return;
     }
+
+    MediaUtil.getSVGAsync(form, markerAsset, new AsyncCallbackPair<SVG>() {
+          @Override
+          public void onFailure(String message) {
+            try {
+              // Passes the defaultMarkerSVG
+              Log.d(TAG, "PASSING DEFAULT");
+              callback.onSuccess(rasterizeSVG(aiMarker, defaultMarkerSVG));
+            } catch (Exception e) {
+              callback.onFailure(e.getMessage());
+            }
+          }
+
+          @Override
+          public void onSuccess(SVG result) {
+            try {
+              Log.d(TAG, "PASSING LOADED result " + (result == null));
+              callback.onSuccess(rasterizeSVG(aiMarker, result));  
+            } catch (Exception e) {
+              e.printStackTrace();
```

Comment:
Use `Log.e` instead of `e.printStackTrace()`

Reply:
Note that you don't log the stack trace in other places, so we should decide whether this is actually needed or not.